functions/db/import_probes.fn.js

```diff
@@ -25,58 +56,88 @@ async function storeProbe(key, probe) {
     .set(probe);
 }
 
+/**
+ * Helper methods.
+ */
+
+function getFirstLastVersion(probe, channel) {
+  return probe.history[channel] && [
+    probe.history[channel].slice(-1)[0].versions.first,
+    probe.history[channel][0].versions.last] || [null, null];
+}
+
+function getOptOut(probe, channel) {
+  if (probe.history[channel]) {
+    return probe.history[channel][0].optout;
+  }
+  return null;
+}
+
 /**
  * Cloud function to fetch probe info and store it in Firestore.
  */
 // eslint-disable-next-line no-multi-assign
 exports = module.exports = functions.https.onRequest((request, response) => {
-  https
-    .get(PROBE_INFO_URL, res => {
-      console.log(res.headers);
-      console.log(res.statusCode);
+  getJSONContent(VERSIONS_URL)
+    .then((versions) => {
+      const [nightly, ] = versions.FIREFOX_NIGHTLY.split('.');
+      console.log(`Latest nightly version: ${nightly}`);
 
-      let data = '';
-      let probe = null;
+      getJSONContent(PROBES_URL)
+        .then((probes) => {
 
-      res.on('data', d => {
-        data += d;
-      });
+          const promises = [];
+          Object.keys(probes).forEach(k => {
+            const probe = probes[k];
+            const oldestHistory = (
```

Comment:
This looks like it's the newest version of the first channel this probe was on. Could you add a comment above this stating that so it's as bit clearer to future me's?

Reply:
Yeah it seems backwards, but `history.nightly[0]` is the most recent set of data and `history.nightly.slice(-1)[0]` is the oldest. I was thinking for the `description`, `kind`, and a few other fields we should get the oldest. But now I'm thinking if they changed maybe we should get the newest values for those.